AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs

```diff
@@ -6,49 +6,96 @@
     using System.Threading.Tasks;
 
     /// <summary>
-    /// Class to contain the one cache for all Gauges 
+    /// Class to contain the one cache for all Gauges.
     /// </summary>
     internal class CacheHelper
     {
+        private static readonly CacheHelper instance = new CacheHelper();
+
+        static CacheHelper() { }
+
+        private CacheHelper() { }
+
+        /// <summary>
+        /// Return the only instance of CacheHelper.
+        /// </summary>
+        public static CacheHelper Instance
+        {
+            get
+            {
+                return instance;
+            }
+        }
+
+        /// <summary>
+        /// Searchs for the value of a given performance counter in a json.
+        /// </summary>
+        /// <param name="performanceCounterName"> The name of the performance counter.</param>
+        /// <param name="json"> String containing the json.</param>
+        /// <returns> Value of the performance counter.</returns>
+        public int PerformanceCounterValue(string performanceCounterName, string json)
+        {
+            string jsonSubstring = json.Substring(json.IndexOf(performanceCounterName), json.Length - json.IndexOf(performanceCounterName));
+
+            int startingIdx = jsonSubstring.IndexOf(" ");
+            int endingIdx = jsonSubstring.IndexOf(",");
+
+            string value = jsonSubstring.Substring(startingIdx + 1, endingIdx - startingIdx - 1);
```

Comment:
It could be the case, if the performance counter isn't found anywhere in the json. I'll validate this.